source/common/http/route_config_update_requster.h

```diff
@@ -0,0 +1,67 @@
+#pragma once
+
+#include "envoy/http/filter.h"
+
+#include "source/common/router/rds_impl.h"
+#include "source/common/router/scoped_rds.h"
+
+namespace Envoy {
+namespace Http {
+
+class RdsRouteConfigUpdateRequester : public RouteConfigUpdateRequester {
+public:
+  RdsRouteConfigUpdateRequester(Router::RouteConfigProvider* route_config_provider)
+      : route_config_provider_(route_config_provider) {}
+
+  RdsRouteConfigUpdateRequester(Config::ConfigProvider* scoped_route_config_provider,
+                                OptRef<const Router::ScopeKeyBuilder> scope_key_builder)
+      // Expect the dynamic cast to succeed because only ScopedRdsConfigProvider is fully
+      // implemented. Inline provider will be cast to nullptr here but it is not full implemented
+      // and can't not be used at this point. Should change this implementation if we have a
+      // functional inline scope route provider in the future.
+      : scoped_route_config_provider_(
+            dynamic_cast<Router::ScopedRdsConfigProvider*>(scoped_route_config_provider)),
+        scope_key_builder_(scope_key_builder) {}
+
+  void requestRouteConfigUpdate(RouteCache& route_cache,
+                                Http::RouteConfigUpdatedCallbackSharedPtr route_config_updated_cb,
+                                absl::optional<Router::ConfigConstSharedPtr> route_config,
+                                Event::Dispatcher& dispatcher,
+                                RequestHeaderMap& request_headers) override;
+  void
+  requestVhdsUpdate(const std::string& host_header, Event::Dispatcher& thread_local_dispatcher,
+                    Http::RouteConfigUpdatedCallbackSharedPtr route_config_updated_cb) override;
+  void
+  requestSrdsUpdate(RouteCache& route_cache, Router::ScopeKeyPtr scope_key,
+                    Event::Dispatcher& thread_local_dispatcher,
+                    Http::RouteConfigUpdatedCallbackSharedPtr route_config_updated_cb) override;
+
+private:
+  Router::RouteConfigProvider* route_config_provider_;
+  Router::ScopedRdsConfigProvider* scoped_route_config_provider_;
+  OptRef<const Router::ScopeKeyBuilder> scope_key_builder_;
+};
+
+class RdsRouteConfigUpdateRequesterFactory : public RouteConfigUpdateRequesterFactory {
```

Comment:
High-level question (OOC): From E-M point-of-view, what is the benefit of having this in the core code and not as an extension?

Reply:
you mean for location?  So I'd been moving things to extensions but we decided because of the review guidelines to keep things built into envoy by default under non-extension locations.  EM doesn't care where the code lives, we just don't want to link it =P